x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/expression.tsx

```diff
@@ -36,8 +35,27 @@ const SearchSourceExpressionMemoized = memo<SearchSourceExpressionProps>(
 export const EsQueryAlertTypeExpression: React.FunctionComponent<
   RuleTypeParamsExpressionProps<EsQueryAlertParams>
 > = (props) => {
-  const { ruleParams, errors } = props;
+  const { ruleParams, errors, setRuleProperty } = props;
   const isSearchSource = isSearchSourceAlert(ruleParams);
+  const [activeQueryFormType, setActiveQueryFormType] = useState<QueryFormType | null>(null);
+
+  const resetFormType = useCallback(() => {
+    // @ts-ignore Reset rule params regardless of their type
```

Comment:
Nit: @ts-expect-error would be preferable, context for this: https://github.com/elastic/kibana/pull/99241